libs/api/database/src/lib/content/schemas/index.ts

```diff
@@ -35,19 +37,24 @@ export async function setupContentCollection() {
     // making a text index on the title field for search
     schema.index({ title: 'text' });
 
-    schema.pre<ContentDocument>('save', async function (next: HookNextFunction) {
-        this.set('title', sanitizeHtml(this.title, sanitizeOptions));
-        this.set('body', sanitizeHtml(this.body, sanitizeOptions));
-
-        // this will only trigger if any creation or editing functions has modified the `desc` field,
-        // otherwise we'll leave it alone
-        if (this.isModified('desc')) {
-            this.set('desc', sanitizeHtml(this.desc, sanitizeOptions));
+    schema.plugin<MongooseFuzzyOptions<ContentDocument>>(registerFuzzySearch,{
+        fields: ['title'],
+        // Middlewares must be defined inside the fuzzy-search plugin, otherwise it will override them.
+        middlewares: {
+            preSave: async function (next: HookNextFunction) {
+                this.set('title', sanitizeHtml(this.title, sanitizeOptions));
+                this.set('body', sanitizeHtml(this.body, sanitizeOptions));
+
+                // this will only trigger if any creation or editing functions has modified the `desc` field,
+                // otherwise we'll leave it alone
+                if (this.isModified('desc')) {
+                    this.set('desc', sanitizeHtml(this.desc, sanitizeOptions));
+                }
+
+                return next();
```

Comment:
Per the mongoose-fuzzy-searching docs, it overrides any `pre()-` hooks you have defined on a schema, so they need to be defined within the plugin's `middlewares` config field. For that reason, I also define it as the first plugin in the chain, to ensure those hooks run as early as possible.